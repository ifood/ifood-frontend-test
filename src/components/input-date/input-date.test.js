import React from 'react'
import { render } from '@testing-library/react'
import { InputDate } from '.'

describe('InputDate', () => {
  let component

  beforeEach(() => {
    const props = {
      id: 'date',
      name: 'date',
      label: 'Date',
      value: null,
      onChange: jest.fn(x => x)
    }

    component = render(<InputDate {...props} />)
  })

  it('should render component correctly', () => {
    const { asFragment } = component
    expect(asFragment()).toMatchSnapshot()
  })
})
