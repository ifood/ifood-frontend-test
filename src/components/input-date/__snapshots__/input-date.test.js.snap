// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputDate should render component correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiFormControl-root makeStyles-formControl-1"
  >
    <div
      class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
        data-shrink="false"
        for="date"
        id="date-label"
      >
        Date
      </label>
      <div
        class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedEnd"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd"
          id="date"
          type="text"
          value=""
        />
        <div
          class="MuiInputAdornment-root MuiInputAdornment-positionEnd"
        >
          <button
            class="MuiButtonBase-root MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
                />
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
